frame/contracts/src/lib.rs

```diff
@@ -356,6 +361,11 @@ pub mod pallet {
 				salt,
 				None,
 			);
+			if let Ok(retval) = &output.result {
+				if !retval.1.is_success() {
```

Comment:
Is it plausible that the implementation of `is_success` changes to check for other flags? If so would it make sense to add a specific `is_reverted` method in addition?

Reply:
I think this new method should replace `is_success`. No reason to have it as long as there is no other failure flag. Better be specific.